server/spinwick.go

```diff
@@ -65,6 +89,103 @@ func (s *Server) handleCreateSpinWick(pr *model.PullRequest, size string, withLi
 			s.logPrettyErrorToMattermost("[ SpinWick ] Creation Failed", pr, request.Error, additionalFields)
 		}
 	}
+
+}
+
+func (s *Server) createKubeSpinWick(pr *model.PullRequest) *spinwick.Request {
+	request := &spinwick.Request{
+		InstallationID: "n/a",
+		Error:          nil,
+		ReportError:    false,
+		Aborted:        false,
+	}
+
+	logger := log.WithField("PR", pr.RepoName+"/!"+string(pr.Number))
+	kc, err := k8s.New(filepath.Join(homedir.HomeDir(), ".kube", "config"), logger)
+
+	namespaceName := makeSpinWickID(pr.RepoName, pr.Number)
+	namespace, err := getOrCreateNamespace(kc, namespaceName)
+
+	if err != nil {
+		request.Error = err
+		return request.WithError(errors.Wrap(err, "Error occurred whilst creating namespace")).ShouldReportError()
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 45*time.Minute)
+	defer cancel()
+
+	version := ""
+	image := "mattermost/cws-test"
+
+	reg, errDocker := s.Builds.dockerRegistryClient(s)
+	if errDocker != nil {
+		return request.WithError(errors.Wrap(errDocker, "unable to get docker registry client")).ShouldReportError()
+	}
+
+	prNew, errImage := s.Builds.waitForImage(ctx, s, reg, pr, image)
+	if errImage != nil {
+		return request.WithError(errors.Wrap(errImage, "error waiting for the docker image. Aborting")).IntentionalAbort()
+	}
+
+	version = s.Builds.getInstallationVersion(prNew)
+
+	deployment := Deployment{
+		namespace.GetName(),
+		version,
+		"/matterwick/templates/cws/cws_deployment" + namespace.GetName() + ".yaml",
+		s.Config.CWS,
+	}
+
+	template, err := template.ParseFiles("/matterwick/templates/cws/cws_deployment.tmpl")
+	if err != nil {
+		mlog.Error("Error loading deployment template ", mlog.Err(err))
+	}
+
+	file, err := os.Create(deployment.DeployFilePath)
+	if err != nil {
+		return request.WithError(errors.Wrap(err, "Error creating deployment file")).ShouldReportError()
+	}
+
+	err = template.Execute(file, deployment)
+	if err != nil {
+		mlog.Error("Error executing template ", mlog.Err(err))
+	}
+	file.Close()
+
+	request.InstallationID = namespace.GetName()
+
+	deployFile := k8s.ManifestFile{
+		Path:            deployment.DeployFilePath,
+		DeployNamespace: deployment.Namespace,
+	}
+	err = kc.CreateFromFile(deployFile, "")
+	if err != nil {
+		return request.WithError(errors.Wrap(err, "Error deploying from manifest template")).ShouldReportError()
+	}
+
+	err = os.Remove(deployment.DeployFilePath)
+
+	if err != nil {
+		return request.WithError(errors.Wrap(err, "Error creating deployment file")).ShouldReportError()
```

Comment:
if we continue with the line above the error message error is not correct. you are removing the file and not creating.

Reply:
Removed based on suggestion above